lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -329,32 +330,36 @@ def run_sync(command, stdin = nil)
       stdout = ''
       stderr = ''
       exit_status = nil
-      started = false
 
-      channel = session.open_channel do |ch|
-        ch.on_data do |c, data|
-          stdout.concat(data)
-        end
-        ch.on_extended_data { |_, _, data| stderr.concat(data) }
-        ch.on_request('exit-status') { |_, data| exit_status = data.read_long }
-        # Send data to stdin if we have some
-        ch.send_data(stdin) unless stdin.nil?
-        # on signal: sending the signal value (such as 'TERM')
-        ch.on_request('exit-signal') do |_, data|
-          exit_status = data.read_string
-          ch.close
-          ch.wait
-        end
-        ch.exec command do |_, success|
-          raise 'could not execute command' unless success
-
-          started = true
-        end
+      master, slave = PTY.open
+      read, write = IO.pipe
+
+      pid = spawn("/usr/bin/sshpass", "-p #{ssh_options[:password]}", "-P #{ssh_options[:passphrase]}",
+                  "/usr/bin/ssh", "#{@ssh_user}@#{@host}",
+                  "-o Port=#{ssh_options[:port].to_s}",
+                  "-o IdentityFile=#{ssh_options[:keys][1..-2]}",
+                  "-o AddKeysToAgent=#{ssh_options[:keys_only]}",
+                  "-o PasswordAuthentication=#{ssh_options[:auth_methods].include? 'password'}",
+                  "-o PubkeyAuthentication=#{ssh_options[:auth_methods].include? 'publickey'}",
```

Comment:
It would probably more future proof if we replace this with `-o PreferredAuthentications=#{ssh_options[:auth_methods].join(',')` to, among other things, express in which order the methods should be tried.